libpdb/topic_test.go

```diff
@@ -4,36 +4,61 @@ import (
 	"crypto/rand"
 	"fmt"
 	"github.com/ryscheng/pdb/common"
-	"strconv"
-	"strings"
 	"testing"
 )
 
 func TestEncryptDecrypt(t *testing.T) {
 	fmt.Printf("TestEncryptDecrypt:\n")
-	password := ""
 	plaintext := "Hello world"
-	nonce := []byte("012345678901")
-	th, err := NewTopic(password)
+	var nonce [24]byte
+	copy(nonce[:],[]byte("012345678901"))
+	th, err := NewTopic()
 	if err != nil {
 		t.Fatalf("Error creating topic handle: %v\n", err)
 	}
-	ciphertext, err := th.Encrypt([]byte(plaintext), nonce)
+	sub, err := th.CreateSubscription()
+	if err != nil {
+		t.Fatalf("Failed to derive subscription from topic: %v\n", err)
+	}
+	ciphertext, err := th.encrypt([]byte(plaintext), &nonce)
 	if err != nil {
 		t.Fatalf("Error encrypting plaintext: %v\n", err)
 	}
-	result, err := th.Decrypt(ciphertext, nonce)
+	result, err := sub.Decrypt(ciphertext, &nonce)
 	if err != nil {
-		t.Fatalf("Error decrypting ciphertext: %v\n", err)
+		t.Fatalf("Error decrypting ciphertext: %v, %v\n", ciphertext, err)
 	}
-	if strings.Compare(plaintext, string(result)) != 0 {
-		t.Fatalf("Invalid decrypted value: %v. Expected:%v\n", string(result), plaintext)
+	if plaintext != string(result) {
```

Comment:
th.createSubscription creates a subscription with the same shared key as the topic.  The subscription object is a different name for the thing you're calling topicHandle, i think - the shared capability to locate and read items.